aws/resource_aws_workspaces_directory.go

```diff
@@ -0,0 +1,145 @@
+package aws
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/workspaces"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceAwsWorkspacesDirectory() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsWorkspacesDirectoryCreate,
+		Read:   resourceAwsWorkspacesDirectoryRead,
+		Update: resourceAwsWorkspacesDirectoryUpdate,
+		Delete: resourceAwsWorkspacesDirectoryDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"directory_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"subnet_ids": {
```

Comment:
Hi @Tensho. Sorry, I wasn't clear enough about what I was suggesting for the subnets. I definitely agree that whoever is using the resource will need to figure out which subnets are compatible, and that will depend on the region that they are using.

However, if they make a change like I suggested above in https://github.com/terraform-providers/terraform-provider-aws/pull/11023#issuecomment-561592697, they will always have changes in their plan. But if you add the `ForceNew` flag to `subnet_ids` as well, this will re-create the `aws_workspaces_directory` resource with the desired subnets. I think this would be the desired behaviour.

Reply:
Ah, I see what you mean. Yes, you are right, any change to `subnet_ids` forces re-creation of workspaces directory because it's not possible to modify it now. I've added `ForceNew`.